### PR TITLE
class library: rebuild proxyspace after copy

### LINKED ISSUE #5180: ProxySpace copy doesn't copy references to currentEnvironment
## Environment

* SuperCollider version:
* Operating system:
* Other details (Qt version, audio driver, etc.):

## Steps to reproduce

From the mailing list:

```supercollider


p = ProxySpace.push(s);
~src.ar(2)
~src = {Blip.ar(LFNoise0.kr(0.3!2).range(5,30), LFNoise1.kr(0.3).range(20,50))}
~hpf = {HPF.ar(~src.ar, \hpf.kr(80))};
// ProxyMixer(p)


o = p.copy; 
p.pop; 
o.push;

// ProxyMixer(o)

~hpf.play
p[\src].pause
p[\src].resume
```

## Expected vs. actual behavior

pausing a proxy in `p` should not influence a proxy in `o`.

### PR DESCRIPTION
**Pull request by telephon**
This fixes  #5180

## Purpose and Motivation

When used within a `ProxySpace`, a NodeProxy's function may contain dynamically scoped references to the local environment. When the `ProxySpace` is copied, the proxies are copied, but they still point to the original sources. This patch updates the references  by rebuilding every proxy from within the new environment. It avoids double rebuilding by keeping a set of those which have been touched already (`checkedAlready`);

## Types of changes

- Bug fix

## To-do list

- [x] Code is tested
- [x] All tests are passing
- [x] This PR is ready for review

### CODE REVIEW

File: testsuite/classlibrary/TestProxySpace.sc [hunk 1]
```diff
@@ -52,4 +60,42 @@ TestProxySpace : UnitTest {
...
+
+
+		copySpace[\hpf].getFamily(family);
+		this.assert(family.size == 2, "there should be one parent in this case", report: true); // sanity check
+
+		commonProxies = proxySpace.envir.values.as(IdentitySet) sect: family;
+		this.assert(commonProxies.isEmpty, "proxyspace copying should refresh local environment crossreferences");
+
+		proxySpace.clear;
+		copySpace.clear;
```
Comment (brianlheim) [inline testsuite/classlibrary/TestProxySpace.sc:94]: i think you forgot to quit the server at the end

**Review by brianlheim — COMMENTED** [review]
thanks! just one comment

File: testsuite/classlibrary/TestProxySpace.sc (hunk 1 above)
Comment (telephon) [reply testsuite/classlibrary/TestProxySpace.sc:92]: ah yes, thanks. I've refactored the boot and quit for both tests.

**Review by brianlheim — APPROVED** [review]
thanks!

**Pull request merged**